### PR TITLE
Added Hover effect for Navbar

### LINKED ISSUE #14: Add hover effect on nav bar links
**Is your feature request related to a problem? Please describe.**
Currently, there is no hover effect on navbar links.

**Describe the solution you'd like**
I would like to have some hover effect on the navbar items. Please refer to the below mentioned codepen resource.

**Additional context**
Apply the last effect `/* Highlighted text in */` mentioned at this [codepen](https://codepen.io/EvyatarDa/pen/waKXMd)

**Do not change**
1. code formatting
2. other files
3. font-size of the nav links

### PR DESCRIPTION
**Pull request by msraghavganesh**
Fixes: #14 

## Proposed changes:
- Added navbar_hover.css
- changed Container styling for navbar

## Screenshots

**After**<br/>
Screenshot (378)

### CODE REVIEW

**Comment by msraghavganesh** [comment]
Suggest any more changes?

File: index.html [hunk 1]
```diff
@@ -25,10 +26,13 @@
...
     <header>
-
       <!-- Github repo link -->
-      <a href="https://github.com/dipanshparmar/sortvisual" id="github-repo-link" target="_blank">
-        <img src="img/Octocat.jpg">
+      <a
+        href="https://github.com/dipanshparmar/sortvisual"
+        id="github-repo-link"
+        target="_blank"
+      >
```
Comment (dipanshparmar) [inline index.html]: revert the formatting changes, please.

File: index.html [hunk 2]
```diff
@@ -37,8 +41,11 @@
       </div>
       <!-- controls with speed and size controlling -->
       <div id="controls">
-        <div id="speed-controls-container">
-          <span>Speed</span>
+        <div
+          id="speed-controls-container"
+          class="container gray highlightTextIn"
+        >
+          <a class="speed button" style="font-size: 16px" alt="Speed">Speed</a>
```
Comment (dipanshparmar) [inline index.html:40]: Why did you change this to the **anchor** tag?  Revert this to span.

File: index.html [hunk 3]
```diff
@@ -37,8 +41,11 @@
       </div>
       <!-- controls with speed and size controlling -->
       <div id="controls">
-        <div id="speed-controls-container">
-          <span>Speed</span>
+        <div
+          id="speed-controls-container"
+          class="container gray highlightTextIn"
+        >
```
Comment (dipanshparmar) [inline index.html]: make it in a single line.

File: index.html [hunk 4]
```diff
@@ -50,8 +57,11 @@
             step="20"
           />
         </div>
-        <div id="size-controls-container">
-          <span>Size</span>
+        <div
+          id="size-controls-container"
+          class="container gray highlightTextIn"
+        >
```
Comment (dipanshparmar) [inline index.html]: make this in a single line.

File: index.html [hunk 5]
```diff
@@ -50,8 +57,11 @@
...
+          <a class="span button" style="font-size: 16px" alt="Size">Size</a>
```
Comment (dipanshparmar) [inline index.html]: Revert it to span and don't change the font size.

File: index.html [hunk 6]
```diff
@@ -65,29 +75,26 @@
...
-        <span class="quick button">Quick</span>
-        <span class="bubble button">Bubble</span>
-        <span class="heap button">Heap</span>
-        <span class="selection button">Selection</span>
+      <div id="options" class="container gray highlightTextIn">
+        <a class="merge button" alt="Merge">Merge</a>
+        <a class="quick button" alt="Quick">Quick</a>
+        <a class="bubble button" alt="Bubble">Bubble</a>
+        <a class="heap button" alt="Heap">Heap</a>
+        <a class="selection button" alt="Selection">Selection</a>
```
Comment (dipanshparmar) [inline index.html:72]: Revert these **anchors** into **span** as well

File: scss/navbar_hover.css [hunk 7]
```diff
@@ -0,0 +1,46 @@
+@import "https://fonts.googleapis.com/css?family=Raleway";
```
Comment (dipanshparmar) [inline scss/navbar_hover.css]: Don't include another font family. We are already using Poppins

File: scss/navbar_hover.css [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+/* Highlight text in  */
+div.highlightTextIn a {
+  color: white;
+}
+
+div.highlightTextIn a:before {
+  transform: scale(1.5);
+}
```
Comment (dipanshparmar) [inline scss/navbar_hover.css]: make all these CSS changes in _header.scss file

File: scss/navbar_hover.css [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  position: relative;
+  z-index: 0;
+  cursor: pointer;
+}
+
+.gray {
+  background: none;
+}
+
+div.highlightTextOut a:before,
```
Comment (dipanshparmar) [inline scss/navbar_hover.css]: remove highlightTextOut as we are only using hightlightTextIn.

File: scss/navbar_hover.css [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  left: 0px;
+  padding: 10px;
+  overflow: hidden;
+  content: attr(alt);
+  transition: all 0.5s;
+  transform: scale(0.8);
+  opacity: 0;
+}
+
+div.highlightTextOut a:hover:before,
```
Comment (dipanshparmar) [inline scss/navbar_hover.css]: remove highlightTextOut here as well

**Review by dipanshparmar — CHANGES_REQUESTED** [review]
Please disable your code formatter while making the changes.

File: index.html (hunk 3 above)
Comment (msraghavganesh) [reply index.html]: I'll disable my code formatter, It should solve this.

File: index.html (hunk 5 above)
Comment (msraghavganesh) [reply index.html]: As we are not using any href attribute, anchor tag works as a span tag itself. Would be useful in future if we add any href attributes. We may not need to edit the css styling when we are adding any href in future. Font Size reverted.

Comment (dipanshparmar) [reply index.html]: I don't think so I have any plans for adding any href here in the near future. So it's better to use the actual tag instead of using some other tag that is meant for some other purposes. Please revert it.

File: scss/navbar_hover.css (hunk 8 above)
Comment (msraghavganesh) [reply scss/navbar_hover.css]: There Seems to be a issue while compiling .scss file, The changes are not being reflected in the live server. For now we will go with external css styling (which works fine), Once the issue is resolved I'll Issue a new PR for the same.

Comment (dipanshparmar) [reply scss/navbar_hover.css]: You need to install **Sass compiler (by Ritwick Dey)**, the VS Code extension to compile the .scss file.

**Comment by dipanshparmar** [comment]
@msraghavganesh Please push the new changes, so that I can see them.

File: index.html (hunk 5 above)
Comment (msraghavganesh) [reply index.html]: Okay, Will do it soon

**Comment by dipanshparmar** [comment]
#28
